yapapi/rest/activity.py

```diff
@@ -182,22 +181,61 @@ def id(self):
 class PollingBatch(Batch):
     """A `Batch` implementation that polls the server repeatedly for command status."""
 
+    async def _activity_terminated(self) -> bool:
```

Comment:
This function does not use `self`. IMO it should be an inner function or a method of `Activity`.

Reply:
What do you mean by "does not use `self`"? It does use `self` to access `self._activity: Activity`.

Reply:
You are right, it's the next method (`_is_endpoint_not_found_error`) that does not use `self`.

Reply:
Also, this can be prefixed with `_is`, i.e. renamed to `_is_activity_terminated` (or become a `is_terminated` method of `Activity`).

Reply:
Moved `_is_endpoint_not_found_error()` out of `PollingBatch` in e5a2b3c